File: clients/asset-client/src/constants/RoleConstant.ts
import { createConstant } from "./Constant";

const RoleConstant = createConstant(
  {
    Admin: "Admin",
    DoanhNghiep: "DoanhNghiep",
    ChuyenVienSo: "ChuyenVienSo",
    TruongPhongSo: "TruongPhongSo",
    LanhDaoSo: "LanhDaoSo",
    ChuyenVienCuc: "ChuyenVienCuc",
    TruongPhongCuc: "TruongPhongCuc",
    LanhDaoCuc: "LanhDaoCuc",
    CVQUANLYTHITRUONG: "CVQUANLYTHITRUONG",
  } as const,
  {
    Admin: { displayName: "Quản trị viên" },
    DoanhNghiep: { displayName: "Doanh nghiệp" },
    ChuyenVienSo: { displayName: "Chuyên viên sở" },
    TruongPhongSo: { displayName: "Trưởng phòng sở" },
    LanhDaoSo: { displayName: "Lãnh đạo sở" },
    ChuyenVienCuc: { displayName: "Chuyên viên cục" },
    TruongPhongCuc: { displayName: "Trưởng phòng cục" },
    LanhDaoCuc: { displayName: "Lãnh đạo cục" },
    CVQUANLYTHITRUONG: { displayName: "Chuyên viên quản lý thị trường" },
  }
);

export default RoleConstant;

File: services/asset-service/Hinet.Service/Constant/RoleConstant.cs
using System.ComponentModel;

namespace Hinet.Service.Constant
{
    public class RoleConstant
    {
        [DisplayName("Quản trị viên")]
        public static string Admin => "Admin";

        [DisplayName("Doanh nghiệp")]
        public static string DoanhNghiep => "DoanhNghiep";

        [DisplayName("Chuyên viên sở")]
        public static string ChuyenVienSo => "ChuyenVienSo";

        [DisplayName("Trưởng phòng sở")]
        public static string TruongPhongSo => "TruongPhongSo";

        [DisplayName("Lãnh đạo sở")]
        public static string LanhDaoSo => "LanhDaoSo";

        [DisplayName("Chuyên viên cục")]
        public static string ChuyenVienCuc => "ChuyenVienCuc";

        [DisplayName("Trưởng phòng cục")]
        public static string TruongPhongCuc => "TruongPhongCuc";

        [DisplayName("Lãnh đạo cục")]
        public static string LanhDaoCuc => "LanhDaoCuc";
        public static string CucTruong => "CucTruong";
        public static string PhoCucTruong => "PhoCucTruong";
        public static string TruongPhong => "TruongPhong";
        public static string PhoTruongPhong => "PhoTruongPhong";

        [DisplayName("Môi giới")]
        public static string MoiGioi => "MoiGioi";

        [DisplayName("Chính chủ")]
        public static string ChinhChu => "ChinhChu";

        [DisplayName("Tìm kiếm")]
        public static string TimKiem => "TimKiem";
    }
}
